File: codeforces/0492C_-_Vanya_and_Exams/main.cpp
#include <bits/stdc++.h>

using namespace std;

typedef long long int ll;

const int MAXN = 1e5;

pair<int, int> p[MAXN];

int main() {
  int n, r, avg;
  scanf("%d %d %d", &n, &r, &avg);

  int points = 0;
  for (int i = 0; i < n; i++) {
    int a, b;
    scanf("%d %d", &b, &a);
    points += a;
    p[i] = {a, b};
  }

  sort(p, p + n);

  int essays = 0;
  int i = 0;
  printf("%d %d\n", points, avg * n);
  while (points < avg * n) {
      int t = min(avg * n - points, r - p[i].second);
      essays += t * p[i].first;
      points += t;
      i += 1;
  }

  printf("%d\n", essays);
}

File: codeforces/0492C_-_Vanya_and_Exams/main.py
def main():
    (n, r, avg) = map(int, input().split(' '))
    x = []
    points = 0
    essays = 0
    for i in range(n):
        a, b = map(int, input().split(' '))
        if a < r:
            x.append([a, b])
        points += a
    x = sorted(x, key=lambda i: i[1])
    i = 0
    while points < int(avg * n):
        t = min(int(avg * n - points), r - x[i][0])
        essays += int(t * x[i][1])
        points += t
        i += 1
    return essays
print(main())
